subprojects/docs/src/samples/java/crossCompilation/groovy/build.gradle

```diff
@@ -8,15 +8,17 @@ repositories {
 
 dependencies {
     // We should use a legacy version to support running on jdk6
-    compile group: 'commons-lang', name: 'commons-lang', version: '2.6'
-    testCompile group: 'junit', name: 'junit', version: '4.+'
+    implementation 'commons-lang:commons-lang:2.6'
+    testImplementation 'junit:junit:4.+'
 }
 
 // tag::java-cross-compilation[]
 assert hasProperty('javaHome'): "Set the property 'javaHome' in your your gradle.properties pointing to a Java 6 or 7 installation"
 assert hasProperty('targetJavaVersion'): "Set the property 'targetJavaVersion' in your your gradle.properties to '1.6' or '1.7'"
 
-sourceCompatibility = targetJavaVersion
+java {
+    sourceCompatibility = JavaVersion.toVersion(targetJavaVersion)
```

Comment:
:x: let's keep the Groovy magic here:
```groovy
java {
    sourceCompatibility = targetJavaVersion
}
```

Reply:
That was my intention, but it doesn't work:

```
Build file '/Users/jb/projects/gradle/subprojects/docs/src/samples/java/crossCompilation/groovy/build.gradle' line: 20

* What went wrong:
A problem occurred evaluating root project 'cross-compilation'.
> Cannot convert string value '1.7' to an enum value of type 'org.gradle.api.JavaVersion' (valid case insensitive values: VERSION_1_1, VERSION_1_2, VERSION_1_3, VERSION_1_4, VERSION_1_5, VERSION_1_6, VERSION_1_7, VERSION_1_8, VERSION_1_9, VERSION_1_10, VERSION_11, VERSION_HIGHER)
```

It works fine when not inside the `java { }` closure though.

Reply:
Ah damn. Then let's keep it as you did.

Reply:
Beware: the exemplar configuration expects the test to fail, so you might get the wrong feeling that the `sourceCompatibility = targetJavaVersion` works fine because the test would be green. It would only be the case because the build fails for a different reason than the expected one (which, I guess, is the absence of a Java 6 / Java 7 JDK at the location specified in the configuration)